sound/soc/sof/ipc.c

```diff
@@ -319,6 +319,16 @@ int sof_client_tx_message(struct device *dev, u32 header,
 }
 EXPORT_SYMBOL(sof_client_tx_message);
 
+/* register client to receive IPC */
+void snd_sof_ipc_rx_register(struct snd_sof_client *client,
+			     struct device *dev)
```

Comment:
it'd be nicer if we have a dev field in snd_sof_client, that way the device information is hidden, and you'd use an indirection client->dev
'dev' alone requires the reviewer to look at the call tree to figure out what the arguments were, it's not that self-explanatory.